openlibrary/plugins/upstream/addbook.py

```diff
@@ -1009,9 +1010,10 @@ def process_input(self, i):
         if 'author' in i:
             author = trim_doc(i.author)
             alternate_names = author.get('alternate_names', None) or ''
-            author.alternate_names = [
-                name.strip() for name in alternate_names.split('\n') if name.strip()
-            ]
+            author.alternate_names = uniq(
+                [name.strip() for name in alternate_names.split('\n') if name.strip()],
+                key=str.lower,
+            )
```

Comment:
Here's a (untested) version that includes the main author name in the deduping.
```suggestion
            author.alternate_names = uniq(
                [author] + 
                [name.strip() for name in alternate_names.split('\n') if name.strip()]
            )[1:]
```

Reply:
I've added this suggestion to the PR.